pkg/reconciler/v1alpha1/revision/revision.go

```diff
@@ -400,6 +400,7 @@ func (c *Reconciler) reconcile(ctx context.Context, rev *v1alpha1.Revision) erro
 		}
 	}
 
+	rev.Status.ObservedGeneration = rev.Generation
```

Comment:
So observed generation for revision exists but we as not being utilized before? Can you add a test as part of this change?

Reply:
I added it to all of our e2e tests, since the fakes don't deal with `metadata.generation` setting :(